### DISCUSSION TITLE
Mesh sample on Xiao SEEED NRF52840 (not sense)

### DISCUSSION THREAD

**Discussion by ShawnRobertson** [zephyrproject-rtos/zephyr#55356] [General]
**Describe the bug**
When attempting to run the Zephyr/Bluetooth/Mesh sample on the Xiao Seeed NRF52840 the device broadcasts Mesh packets however appears to provision itself with a Network ID of 0x00, then 0x01, and so on.  It does not broadcast a mesh network name but most definitely broadcasts mesh packets.  The attached image shows, in the NRF connect app, the packets being broadcast by the device.  It does not allow for provisioning however.   I found in the main.c for the sample a self provisioning call when a button is pushed and attempted to fix this by commenting out that section, assuming that somehow the device was sending a true on a button push despite the device having only a reset button.  This did not work.  I have been able to run the sample on an NRF528040 DK from makerdiary.   

![Screenshot_20230227_145256_nRF Connect](https://user-images.githubusercontent.com/89648588/221686776-e9dea901-6169-4fa3-842a-afdab1fddc91.jpg)

Please also mention any information which could help others to understand
the problem you're facing:
 - What target platform are you using: Xiao Seeed NRF52840
 - What have you tried to diagnose or workaround this issue: Code manipulation on the main.c and examination of the included board files as well as using the sample on other boards.
 - Is this a regression? no

**To Reproduce**
We are using nrfConnect for VS Code to build a UF2 file and drag and drop to the device.
1. Create a new application with nRF SDK 2.2 and Toolchain 2.2
2. Select the  zephyr/samples/bluetooth/mesh sample
3. Create application
4. Select the xiao_ble board and the prj.conf config file. 
5. Drag and drop to xiao seeed device 

**Expected behavior**
Capability to provision the device to a ble mesh network.  

**Impact**
Showstopper

**Environment (please complete the following information):**
 - OS: Windows
 - Toolchain: NRF SDK 2.2 Zephyr
 - Zephyr version 2.2

**Comment by PavelVPV** [comment]
Hi @ShawnRobertson, this is true, the sample provisions itself upon button press. To disable this, comment out this code: https://github.com/zephyrproject-rtos/zephyr/blob/8b4407ab7cfb8319c3bf57cb01cbcb8cf9b7eb46/samples/bluetooth/mesh/src/main.c#L335-L374, then erase the device (so that it doesn't restore the previously stored provisioning data) and flash the sample again.

**Comment by ShawnRobertson** [comment]
Hi @PavelVPV, thanks for your response.  We were able to get the code working on a new device however are unable to deprovision a device once it is provisioned once.  We utilized the nRF mesh app to provision the device to test, then deprovisioned it based on the apps guidance, however it did not actually deprovision as we can still see it broadcasting secure packets associated with the provisioned network.  We tried reflashing the sample as well as reburning the bootloader.  Neither worked.  Do you know of a way to completely wipe these?  I know this is more of a question for  Seeed but was hoping you might know.

**Comment by carlescufi** [comment]
@ShawnRobertson @PavelVPV converted to a discussion since this is not  a Zephyr bug.

**Comment by PavelVPV** [comment]
Have you tried to erase the whole flash? Did you unprovisioned the device by pressing on the "Reset" button in the app? And after that, if you press on "ADD NODE" button, do you see the device in the appeared Scanner window?

**Reply by ShawnRobertson** [reply]
We tried through the reset button in the app however I believe it did not erase all of the memory locations and the mesh firmware is still referencing that when reloaded.  When we hit "ADD NODE" it doesn't show back up and we are left with a device that only broadcasts secure packets on the old application keys.

**Reply by PavelVPV** [reply]
Can you enable Persistent settings, Foundation model and Access layer logs in mesh and repeat the actions you do? For nRF Connect SDK 2.2 it should be `CONFIG_BT_MESH_DEBUG_SETTINGS`, `CONFIG_BT_MESH_DEBUG_MODEL`, `CONFIG_BT_MESH_DEBUG_ACCESS`.

**Comment by HerbertDobernig** [comment]
I had the same issue as Shawn had.
I solved this by temporary changing to "CONFIG_SETTINGS=n" in the config file "prj.conf".
(this disables the call to "settings_load()" in "main.c" keeping the Xiao Seeed NRF52840 unprovisioned during startup)

**Reply by HerbertDobernig** [reply]
I did find a more effective way to clean up the Xiao Seeed NRF52840 by writing a "clean-up program" containing this code:

    settings_delete("bt/hash");
    settings_delete("bt/mesh/Seq");
    settings_delete("bt/mesh/IV");
    settings_delete("bt/mesh/Net");
    settings_delete("bt/mesh/NetKey/0");
    settings_delete("bt/mesh/Net");
    settings_delete("bt/mesh/NetKey/0");
    settings_delete("bt/mesh/AppKey/0");
    settings_delete("bt/mesh/s/2/bind");
    settings_delete("bt/mesh/s/3/bind");
    settings_delete("bt/mesh/RPL/1");  
    settings_delete("bt/mesh/RPL/20");
    settings_delete("bt/mesh/RPL/21");

Any other remaining keys can be found by using "guiconfig" ("west build -t guiconfig"):
Subsystems and OS Services -> Settings -> Max compiled-in log level for settings -> Debug
During execution of the function "settings_load()" the keys are print to the console
[00:00:05.271,667] <dbg> settings: settings_call_set_handler: set-value OK. key: bt/mesh/s/3/bind
[00:00:05.271,881] <dbg> settings: settings_call_set_handler: set-value OK. key: bt/mesh/s/2/bind
[00:00:05.273,498] <dbg> settings: settings_call_set_handler: set-value OK. key: bt/mesh/AppKey/0
...
